overlord/snapstate/autorefresh.go

```diff
@@ -797,6 +810,37 @@ func inhibitRefresh(st *state.State, snapst *SnapState, snapsup *SnapSetup, info
 	return busyErr
 }
 
+func processInhibitedAutoRefresh(chg *state.Change, old state.Status, new state.Status) {
+	if !new.Ready() || chg.Kind() != "auto-refresh" {
+		return
+	}
+
+	if err := maybeAddRefreshInhibitNotice(chg.State()); err != nil {
+		logger.Debugf(`internal error: failed to add "refresh-inhibit" notice: %v`, err)
+		return
+	}
+}
+
+func maybeAddRefreshInhibitNotice(st *state.State) error {
+	snapStates, err := All(st)
+	if err != nil {
+		return err
+	}
+
+	for _, snapst := range snapStates {
```

Comment:
this ultimately would need to remember the set of snaps, and emit only if the set has changed? has we don't have RepeatCheck anymore we need anyway to roll something here explicitly, the PR is small enough that we should do that now

Reply:
updated, thanks!

Reply:
I went with a simple mechanism of saving the last set of inhibited snaps in state in `last-recorded-inhibited-snaps`. I don't have a strong preference on the key name.